packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -243,6 +249,24 @@ export class PipelineCreateReplaceChangeSetAction extends PipelineCloudFormation
     });
 
     this.addInputArtifact(props.templatePath.artifact);
+    if (props.templateConfiguration && props.templateConfiguration.artifact.name !== props.templatePath.artifact.name) {
+      this.addInputArtifact(props.templateConfiguration.artifact);
+    }
+
+    const stackArn = cdk.ArnUtils.fromComponents({ service: 'cloudformation', resource: 'stack', resourceName: `${props.stackName}/*` });
```

Comment:
I think it makes sense to have a separate function for this, something like `cfnStackResourcesArn` (this expression is repeated in 2 places).

Reply:
Yah this makes sense :)